From 5d90c4e7be7a00b3c05aad21cf18b494f89349b0 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 19 Apr 2022 10:43:55 +0000
Subject: [PATCH] webosscreenshot: respect QT_OPENGL_ES*

* make sure to include the same header files as QtGui/qopengl.h
  otherwise there is a mismatch in glShaderSource() when
  webosscreenshot includes GLES2/gl2.h and qopengl.h includes
  GLES3/gl32.h e.g. on tissot, tenderloin

In file included from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/git/modules/weboscompositor/webosscreenshot.cpp:22:
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/GLES2/gl2.h:572:37: error: conflicting declaration of C function 'void glShaderSource(GLuint, GLsizei, const GLchar**, const GLint*)'
  572 | GL_APICALL void         GL_APIENTRY glShaderSource (GLuint shader, GLsizei count, const GLchar** string, const GLint* length);
      |                                     ^~~~~~~~~~~~~~
In file included from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/QtGui/qopengl.h:105,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/QtQuick/qsggeometry.h:44,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/QtQuick/qsgnode.h:43,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/QtQuick/qsgrendererinterface.h:43,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/QtQuick/qquickwindow.h:44,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/QtQuick/QQuickWindow:1,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/git/modules/weboscompositor/webosscreenshot.h:23,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/git/modules/weboscompositor/webosscreenshot.cpp:17:
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/aarch64-halium-webos-linux/luna-surfacemanager/1.0-r0/recipe-sysroot/usr/include/GLES3/gl32.h:621:29: note: previous declaration 'void glShaderSource(GLuint, GLsizei, const GLchar* const*, const GLint*)'
  621 | GL_APICALL void GL_APIENTRY glShaderSource (GLuint shader, GLsizei count, const GLchar *const*string, const GLint *length);
      |                             ^~~~~~~~~~~~~~

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
 modules/weboscompositor/webosscreenshot.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/modules/weboscompositor/webosscreenshot.cpp b/modules/weboscompositor/webosscreenshot.cpp
index 9b43c3b..532adfa 100644
--- a/modules/weboscompositor/webosscreenshot.cpp
+++ b/modules/weboscompositor/webosscreenshot.cpp
@@ -19,8 +19,15 @@
 #include "weboscompositortracer.h"
 #include "weboscompositorwindow.h"
 
-#include <GLES2/gl2.h>
-#include <GLES2/gl2ext.h>
+#  if QT_CONFIG(opengles32)
+#   include <GLES3/gl32.h>
+#  elif QT_CONFIG(opengles31)
+#   include <GLES3/gl31.h>
+#  elif QT_CONFIG(opengles3)
+#   include <GLES3/gl3.h>
+#  else
+#   include <GLES2/gl2.h>
+#  endif
 
 #include <QImage>
 #include <QGuiApplication>
